sources/ClangSharp/Wrappers/CXDiagnostic.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+
+namespace ClangSharp
+{
+    public unsafe partial struct CXDiagnostic
+    {
+        public CXDiagnostic(IntPtr pointer)
+        {
+            Pointer = pointer;
+        }
+
+        public IntPtr Pointer;
+
+        public static explicit operator CXDiagnostic(void* value)
```

Comment:
Using explicit operators to convert from the native type to the wrapper type when it is an "ambiguous" match.